apps/server/src/modules/news/controller/news.controller.ts

```diff
@@ -1,41 +1,47 @@
 import { Controller, Get, Post, Body, Param, UseInterceptors, ClassSerializerInterceptor, Query } from '@nestjs/common';
-import { NewsService } from '../news.service';
+import { NewsUc } from '../uc/news.uc';
 import { ApiTags } from '@nestjs/swagger';
 import { Authenticate, CurrentUser } from '../../authentication/decorator/auth.decorator';
 import { ICurrentUser } from '../../authentication/interface/jwt-payload';
 import { ParseObjectIdPipe } from '../../../shared/core/pipe/parse-object-id.pipe';
-import { Types } from 'mongoose';
-import { News } from '../entity/news.entity';
-import { CreateNewsDto } from './dto/news.dto';
+import { CreateNewsRequestDto, NewsResponseDto } from './dto';
 import { PaginationQueryDto } from '../../../shared/core/controller/dto/pagination.query.dto';
+import { NewsMapper } from '../mapper/news.mapper';
 
 @ApiTags('News')
 @Authenticate('jwt')
 @Controller('news')
 @UseInterceptors(ClassSerializerInterceptor) // works only for class instances, default object are not covered!
 export class NewsController {
-	constructor(private readonly newsService: NewsService) {}
+	constructor(private readonly newsUc: NewsUc) {}
 
 	@Post()
-	create(@Body() createNewsDto: CreateNewsDto): Promise<News> {
-		return this.newsService.create(createNewsDto);
+	async create(@Body() createNewsDto: CreateNewsRequestDto): Promise<NewsResponseDto> {
+		const news = await this.newsUc.create(NewsMapper.mapCreateNewsToDomain(createNewsDto));
+		const dto = NewsMapper.mapToResponse(news);
+		return dto;
 	}
 
 	@Get()
-	findAll(@CurrentUser() currentUser: ICurrentUser, @Query() pagination: PaginationQueryDto): Promise<News[]> {
-		return this.newsService.findAllForUser(currentUser, pagination);
+	async findAll(
+		@CurrentUser() currentUser: ICurrentUser,
+		@Query() pagination: PaginationQueryDto
+	): Promise<NewsResponseDto[]> {
+		const newsList = await this.newsUc.findAllForUser(currentUser.userId, pagination);
+		const dtoList = newsList.map((news) => NewsMapper.mapToResponse(news));
```

Comment:
We should ensure that the global serializer not Response Data that does not match the given return type.